documents/src/pages/elements/datetime-picker.md

```diff
@@ -343,6 +344,287 @@ section {
+
+The example below listens to `before-cell-render` event to query slot contents and uses state from `cell` to add CSS classes to the slot content.
+
+```html
+<ef-calendar fill-cells view="2023-04" lang="de">
+  <div class="custom-cell" slot="2023-04-04"></div>
+  <div class="custom-cell" slot="2023-04-24"></div>
+  <div class="custom-cell" slot="2023-04-28"></div>
+  <div class="custom-cell" slot="2023-05-16"></div>
+  <div class="custom-cell" slot="2023-05-25"></div>
+  <div class="custom-cell" slot="2023-05-31"></div>
+  <div class="custom-cell" slot="2023-04"></div>
+  <div class="custom-cell" slot="2023-05"></div>
+  <div class="custom-cell" slot="2023"></div>
+</ef-calendar>
+```
+
+```javascript
+const datetimePicker = document.querySelector('ef-datetime-picker');
+
+datetimePicker?.addEventListener('before-cell-render', (event) => {
+  const sourceDatetimePicker = event.target;
+  const { cell, calendar } = event.detail;
+  const id = calendar.id === 'calendar-to' ? 'to-' : 'from-';
+  const customCell = sourceDatetimePicker.querySelector(`[slot="${id}${cell.value}"]`);
+  
+  // skip style overriding if there is no content for the cell
+  if (!customCell) { return; }
+
+  // use text from component as calendar has built-in locale support
+  // for instance, Mai instead of May in German
+  customCell.textContent = cell.text;
+
+  // modify classes that match to current cell state
+  const customCellClass = customCell.classList;
+  const keys = ['range', 'selected'];
+  for (const key of keys) {
+    cell[key] ? customCellClass.add(key) : customCellClass.remove(key);
+  }
+});
+```
+
+```typescript
+import { BeforeCellRenderEvent, CalendarCell } from '@refinitiv-ui/elements/calendar';
+import { DatetimePicker } from '@refinitiv-ui/elements/datetime-picker';
+
+const datetimePicker = document.querySelector<DatetimePicker>('ef-datetime-picker');
+
+datetimePicker?.addEventListener('before-cell-render', (event) => {
+  const sourceDatetimePicker = event.target as DatetimePicker;
+  const { cell, calendar } = (event as BeforeCellRenderEvent).detail;
+  const id = calendar.id === 'calendar-to' ? 'to-' : 'from-';
+  const customCell = sourceDatetimePicker.querySelector(`[slot="${id}${cell.value}"]`);
+
+  // skip style overriding if there is no content for the cell
+  if (!customCell) { return; }
+  
+  // use text from component as calendar has built-in locale support
+  // for instance, Mai instead of May in German
+  customCell.textContent = cell.text;
+
+  // modify classes that match to current cell state
+  const customCellClass = customCell.classList;
+  const keys = ['range', 'selected'] as Array<keyof CalendarCell>;
+  for (const key of keys) {
+    cell[key] ? customCellClass.add(key) : customCellClass.remove(key);
+  }
+});
+```
+
+```css
+ef-datetime-picker .custom-cell {
+  ...
+}
+ef-datetime-picker .custom-cell.range {
+  ...
+}
+ef-datetime-picker .custom-cell.select {
+  ...
+}
+```
+
+::
```

Comment:
Calendar overlay covers most of the list of Germany economic events. Maybe the list should be placed above of datetime-picker instead?
![image](https://github.com/Refinitiv/refinitiv-ui/assets/117723407/cc0e73db-70a1-49c5-b264-980d5d5bfd1f)

Reply:
I already swap the list to right side.